### ISSUE TITLE
Can't import JS script as JS --> functions lost

### ISSUE THREAD

**Issue by jaustin** [Microsoft/pxt#470]
When choosing "Open" from the main bar:
![image](https://cloud.githubusercontent.com/assets/147113/19303598/3607d05e-9061-11e6-9cc9-0f0fd3ed0a1d.png)
neither

* the current JS/Blocks mode
* nor the mode the script was programmed in

appears to be respected. As a result, when re-loading javascript programs, such as the one attached, only the blocks representation gets shown. This removes JS code.

I haven't verified that the whole JS code is actually being saved into the hex, but the result is this:

Write JS using functions --> Download onto micro:bit --> re-upload into PXT --> Can't re-create original code.

The example attached is generated from the following code:
```
basic.showNumber(0)
show_other()

function show_other() {
    basic.showNumber(1)
}
```

After re-importing and switching back to JS I see this:
```
basic.showNumber(0)
```
[microbit-JSBlocksReproducer.hex.zip](https://github.com/Microsoft/pxt/files/523946/microbit-JSBlocksReproducer.hex.zip)

I think that when opening a script
* If the script was 'compiled' from blocks, then it should be opened in blocks
* If the script was 'compiled' from js (IE the user was in JS mode when 'download' was pressed) then it should be loaded as JS.

If the following actions result in the current mode of the editor changing, the user should be prompted with something non-modal, for example:
![image](https://cloud.githubusercontent.com/assets/147113/19303812/0cc48d80-9062-11e6-998a-6634fff1e062.png)

**Comment by pelikhan** [comment]
@microsoftsam started working on this yesterday.

**Comment by jaustin** [comment]
Just a question on the fix, looking at #481 it doesn't appear to give any feedback about editor mode switching - shall I file that somewhere else as a feature request?

**Comment by pelikhan** [comment]
Yes, please split the issue.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________
